### PR TITLE
Add missing timer IRQHandler and fix wrong timer IRQHandler

### PR DESCRIPTION
**Pull request by fprwi6labs**
Fix issue #111

### CODE REVIEW

File: cores/arduino/stm32/timer.c [hunk 1]
```diff
@@ -1029,199 +1029,343 @@ void attachIntHandle(stimer_t *obj, void (*irqHandle)(stimer_t *))
...
 /******************************************************************************/
 
+#if defined(TIM1_BASE)
 /**
-  * @brief  TIM1 & TIM10 irq handler
+  * @brief  TIM1 IRQHandler
   * @param  None
   * @retval None
   */
+#if defined(STM32F0xx)
```
Comment (fpistm) [inline cores/arduino/stm32/timer.c]: Why not use TIM1_IRQHandler and define the proper one like the other (#define TIM1_IRQHandler TIM1_BRK_UP_TRG_COM_IRQHandler ...) in timer.h
in order to be homogeneous?

File: cores/arduino/stm32/timer.c [hunk 2]
```diff
@@ -1029,199 +1029,343 @@ void attachIntHandle(stimer_t *obj, void (*irqHandle)(stimer_t *))
...
+void TIM1_UP_TIM16_IRQHandler(void)
+#elif defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
 void TIM1_UP_TIM10_IRQHandler(void)
+#endif
 {
   if(timer_handles[0] != NULL) {
     HAL_TIM_IRQHandler(timer_handles[0]);
   }
 
+#if defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
```
Comment (fpistm) [inline cores/arduino/stm32/timer.c]: So this one could be removed as handle by TIM10_IRQHandler

File: cores/arduino/stm32/timer.c [hunk 3]
```diff
@@ -1029,199 +1029,343 @@ void attachIntHandle(stimer_t *obj, void (*irqHandle)(stimer_t *))
...
   if(timer_handles[9] != NULL) {
     HAL_TIM_IRQHandler(timer_handles[9]);
   }
+#endif
+
+#if defined(STM32F1xx) || defined(STM32F3xx) || defined(STM32L4xx)
```
Comment (fpistm) [inline cores/arduino/stm32/timer.c]: So this one could be removed as handle by TIM16_IRQHandler

File: cores/arduino/stm32/timer.c (hunk 1 above)
Comment (fprwi6labs) [reply cores/arduino/stm32/timer.c]: I tried but when timer 10 or 16 are present, there is an error about redefinition:
'error: redefinition of 'TIM1_UP_TIM16_IRQHandler'

Comment (fpistm) [reply cores/arduino/stm32/timer.c]: On which MCU ?

however, for TIM10 handler, it seems strange for STM32F1xx
```
+#elif defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
 void TIM1_UP_TIM10_IRQHandler(void)
```
but 
```
+#if defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
   if(timer_handles[9] != NULL) {
     HAL_TIM_IRQHandler(timer_handles[9]);
```

**Pull request merged**